### PR TITLE
Initial support for scaffolding config in the pipeline execution tab

### PR DESCRIPTION
**Pull request by bengotow**
This PR makes it so that if you 1) open the execution tab for the first time with no saved pipelines or 2) click "Add..." and create a new config yaml, the content of the document defaults to a nice scaffold.

The scaffold includes a header note and definitions for all required fields that are `isOptional:false` in the field tree. Right now, I'm inserting some bogus default values so it's easy to see what type of value you need to provide (For example, `path: "/file/path/here"`), but in the future I'd really like to make these higlighted / templatized in some way.

### CODE REVIEW

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import { ApolloClient } from "apollo-boost";
+import { ValidationResult } from "./codemirror-yaml/mode";
+import { ConfigEditorPipelineFragment } from "./types/ConfigEditorPipelineFragment";
+import {
+  ConfigEditorCheckConfigQuery,
+  ConfigEditorCheckConfigQueryVariables
+} from "./types/ConfigEditorCheckConfigQuery";
+import * as YAML from "yaml";
+
+export const CONFIG_EDITOR_PIPELINE_FRAGMENT = gql`
```
Comment (bengotow) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: Previously the Config Editor made it's own GraphQL query for this data. Now this fragment is exported by the config editor module and we incorporate it into the main pipeline query made by the execute tab. The helper methods in this file operate on this fragment and transform it into 1) the data required for the `<ConfigEditor>` and 2) the YAML scaffold. I'm not sure if there's a best practice for free-standing GraphQL fragments like this - it's not really "owned" by the ConfigEditor component anymore, it's just the data that is required for the helpers here.

**Comment by schrockn** [comment]
screenshot 2019-01-08 13 12 07

So in testing here when there is a "selector" it fills out all the values. Only one of these will be used. We could do that as long as there is a comment that indicates that? Interested to hear @mgasner 's thoughts

**Comment by schrockn** [comment]
Let's leave all but the first entry of the selector commented out @bengotow thx @mgasner for the suggestion

**Comment by schrockn** [comment]
https://github.com/dagster-io/dagster/issues/565

**Pull request merged**